compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1322,8 +1322,19 @@ object Parsers {
       def functionRest(params: List[Tree]): Tree =
         atSpan(start, accept(ARROW)) {
           val t = typ()
+
           if (imods.isOneOf(Given | Erased)) new FunctionWithMods(params, t, imods)
-          else Function(params, t)
+          else if (ctx.settings.YkindProjector.value) {
+            val (newParams :+ newT, tparams) = replaceKindProjectorPlaceholders(params :+ t)
+
+            if (tparams.isEmpty) {
+              Function(params, t)
+            } else {
+              LambdaTypeTree(tparams, Function(newParams, newT))
+            }
```

Comment:
I think this is equivalent to:
```scala
lambdaAbstract(tparams, Function(newParams, newT))
```
(lambdaAbstract could also be used in a couple of other places)

Reply:
I was using `LambdaTypeTree` directly to avoid using the new params, etc. when they weren't needed, and in the tuple case it lets us avoid instantiating a new `Tuple` value:

```scala
              if (tparams.isEmpty) {
                t
              } else {
                LambdaTypeTree(tparams, Tuple(newParams))
              }
```
I'm changing the other two now, and am happy to change the tuple one as well if you'd prefer.